### ISSUE TITLE
The space between legend and graph is too large for a holomap of Overlays

### ISSUE THREAD

**Issue by ruoyu0088** [holoviz/holoviews#3786]
Here is the code:

```
import holoviews as hv
hv.extension("bokeh")

import numpy as np

hm = hv.HoloMap(kdims=["freq"])

t = np.linspace(0, 3, 200)

for freq in [1, 2, 3]:
    hm[freq] = (hv.Curve((t, np.sin(2 * np.pi * freq * t)) , label="sin") * 
                hv.Curve((t, np.cos(2 * np.pi * freq * t)) , label="cos"))
    
hm.opts(hv.opts.Overlay(legend_position="top"))
```

![image](https://user-images.githubusercontent.com/826871/59828357-cb227300-9375-11e9-975b-5d00efcc4637.png)

If only show one overlay:

```
hm.opts(hv.opts.Overlay(legend_position="top"))[1]
```

![image](https://user-images.githubusercontent.com/826871/59828425-f907b780-9375-11e9-8c7b-fe336ce806af.png)

**Comment by kkumawat25** [comment]
Even I am facing same issue. Trick use dynamicMap instead of HoloMap.

**Comment by philippjfr** [comment]
Fixed in #4249.
